src/adapter.rs

```diff
@@ -240,16 +240,17 @@ unsafe fn get_adapter(adapter_addresses_ptr: PIP_ADAPTER_ADDRESSES) -> Result<Ad
 }
 
 unsafe fn socket_address_to_ipaddr(socket_address: &SOCKET_ADDRESS) -> IpAddr {
-    let sockaddr = socket2::SockAddr::from_raw_parts(
-        socket_address.lpSockaddr as *const SOCKADDR,
-        socket_address.iSockaddrLength,
-    );
-
-    // Could be either ipv4 or ipv6
-    sockaddr
-        .as_inet()
-        .map(|s| IpAddr::V4(*s.ip()))
-        .unwrap_or_else(|| IpAddr::V6(*sockaddr.as_inet6().unwrap().ip()))
+    let (_, sockaddr) = socket2::SockAddr::init(|storage, length| {
+        debug_assert!(
+            std::mem::size_of_val(&*socket_address.lpSockaddr) <= std::mem::size_of_val(&*storage)
+        );
+        storage.copy_from_nonoverlapping(socket_address.lpSockaddr as *const _, 1);
```

Comment:
I think this line will copy only 16 bytes (the size of `SOCKADDR`) but it won't work if the socket is ipv6.
I think exactly `socket_address .iSockaddrLength` bytes should be copied here. Am I right?

Reply:
I've changed it to copy `socket_address.iSockaddrLength` bytes.